pkg/virt-operator/webhooks/kubevirt-update-admitter.go

```diff
@@ -162,3 +181,104 @@ func validateCertificates(certConfig *v1.KubeVirtSelfSignConfiguration) []metav1
 
 	return statuses
 }
+
+func validateWorkloadPlacement(namespace string, placementConfig *v1.NodePlacement, client kubecli.KubevirtClient) []metav1.StatusCause {
+	statuses := []metav1.StatusCause{}
+
+	config, err := util.GetConfigFromEnv()
+	if err != nil {
+		statuses = append(statuses, metav1.StatusCause{
+			Type:    metav1.CauseTypeUnexpectedServerResponse,
+			Message: err.Error(),
+		})
+		return statuses
+	}
+
+	handler, err := components.NewHandlerDaemonSet(
+		config.GetNamespace(),
+		config.GetImageRegistry(),
+		config.GetImagePrefix(),
+		config.GetHandlerVersion(),
+		config.GetLauncherVersion(),
+		"productName",
+		"productVersion",
+		config.GetImagePullPolicy(),
+		config.GetVerbosity(),
+		config.GetExtraEnv(),
```

Comment:
unfortunately this isn't always accurate. It's possible to install a version of KubeVirt that doesn't match the version of the operator that is currently installed. 

So getting the config from the environment varialbes of the running virt-operator won't always represent what is being installed.

For example, take a look at this line, https://github.com/kubevirt/kubevirt/blob/78e64e19bed9411bd11cb543f18e6ff551e34a7d/pkg/virt-operator/kubevirt.go#L1023 


We're creating the virt-handler daemonset based on the installstrategy for the version of kubevirt being installed, not based on the running virt-operators envvars.

Reply:
@davidvossel Hi. Since using the install strategy in an accurate manner will be a heavy-lifting, we've decided to fall back to the usage of mocked deployment/DS.
Can you PTAL at the new changes?